### PR TITLE
Update deactivate stake logic

### PR DESCRIPTION
**Pull request by febo**
This PR updates the logic on the `DeactivateStake`:
- Cancel deactivation when `amount` is `0`
- Use `u128` in the deactivation amount calculation
- Use the `Stake` amount to calculate the deactivation ratio

It also adds validation for the `max_deactivation_basis_points` value on `InitializeConfig` and `UpdateConfig`.

### CODE REVIEW

File: program/src/processor/deactivate_stake.rs [hunk 1]
```diff
@@ -12,13 +12,16 @@ use crate::{
     state::{find_stake_pda, Config, Stake},
 };
 
+const MAX_BASIS_POINTS: u128 = 10_000;
```
Comment (joncinque) [inline program/src/processor/deactivate_stake.rs]: side note: unless I'm missing something, there's no check during initialize config or update config that the `basis_points <= 10_000`. we should add that in

**Review by joncinque — APPROVED** [review]
Looks great, thanks! Noticed another thing to fix :sweat_smile:

**Review by buffalojoec — APPROVED** [review]

File: program/src/processor/deactivate_stake.rs (hunk 1 above)
Comment (febo) [reply program/src/processor/deactivate_stake.rs]: Oh, good catch, definitely need to add that.

Comment (febo) [reply program/src/processor/deactivate_stake.rs]: Done.

File: program/src/processor/deactivate_stake.rs [hunk 2]
```diff
@@ -107,14 +108,10 @@ pub fn process_deactivate_stake(
...
-    require!(amount > 0, StakeError::InvalidAmount);
-
     require!(stake.amount >= amount, StakeError::InsufficientStakeAmount);
 
-    let max_deactivation_amount = get_max_deactivation_amount(
-        config.token_amount_delegated,
-        config.max_deactivation_basis_points,
-    )?;
+    let max_deactivation_amount =
+        get_max_deactivation_amount(stake.amount, config.max_deactivation_basis_points)?;
```
Comment (joncinque) [inline program/src/processor/deactivate_stake.rs:114]: oops, I didn't even realize this was incorrect before, nice fix!

**Review by joncinque — APPROVED** [review]

File: program/src/processor/deactivate_stake.rs (hunk 2 above)
Comment (febo) [reply program/src/processor/deactivate_stake.rs:114]: Yeah, it was something that we picked up with Joe when we went through the docs again last week. 😀

**Pull request merged**